oemof_b3/tools/testing_pipeline.py

```diff
@@ -163,83 +181,169 @@ def get_abs_path_list(output_rule_list):
 
     Outputs
     -------
-    absolute_path_list : str
-        Absolute file path
+    absolute_path_list : list
+         Absolute file path in list
 
     """
-    # Loop over each rule which is tested in the snakemake pipeline
-    for sublist in output_rule_list:
-        # Get absolute path
-        absolute_path_list = [os.path.join(os.getcwd(), entry) for entry in sublist]
+    # Get absolute path of rule
+    absolute_path_list = [os.path.abspath(entry) for entry in output_rule_list]
 
     return absolute_path_list
 
 
+def file_name_extension(raw_file_path):
+    """
+    This function rearranges the current absolute file path
+    with the new extension suffix '_original'.
+
+    Inputs
+    -------
+    raw_file_path : str
+        Absolute path of rule
+
+    Outputs
+    -------
+    renamed_path : str
+
+    """
+    # Get file extension
+    file_extension = raw_file_path[raw_file_path.rfind(".") + 1 :]
+    # Rename existing user data
+    renamed_file = rename_path(
+        raw_file_path, "." + file_extension, "_original." + file_extension
+    )
+
+    return renamed_file
+
+
+def rule_test(sublist):
+    """
+    This function runs the rule from the output rule sublist.
+
+    Inputs
+    -------
+    sublist : str
+        Path of rule
+
+    Outputs
+    -------
+    None
+
+    """
+    # Run the snakemake rule in this loop
+    output = snakemake.snakemake(
+        targets=sublist,
+        snakefile="Snakefile",
+    )
+
+    # Check if snakemake rule exited without error (true)
+    assert output, f"Snakemake rule failed for targets: {sublist}"
+
+    # Log the success
+    logging.info(f"Snakemake rule executed successfully for targets: {sublist}")
+
+
+def clean_file(sublist, delete_switch, renamed_path):
+    """
+    This function removes test data files and reverts renamed files.
+
+    Inputs
+    -------
+    sublist : list of str
+        List of target paths of rules
+    delete_switch : bool
+        If True, delete the data created during the test run.
+        If False, do not delete the data.
+    renamed_path : list of str
+        List of renamed target paths
```

Comment:
For consistency reasons:

    renamed_file_path_list : list of str
        List with renamed absolute file path